src/kontrol/utils.py

```diff
@@ -25,6 +27,29 @@
 _LOGGER: Final = logging.getLogger(__name__)
 
 
+def ensure_name_is_unique(name: str, config: KInner) -> str:
+    """Ensure that a given name for a KVariable is unique within the context of a CTerm.
+
+    :param name: name of a KVariable
+    :param cterm: cterm
+    :return: Returns the name if it's not used, otherwise appends a suffix.
+    :rtype: str
+    """
+    new_config = Subst({name: token(True)})(config)
```

Comment:
Do we not have `free_vars` or something like that to figure out the variables we have? Isn't it easier to use that and then examine the set?

Reply:
Yup, I've updated the function to use `free_vars`